pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/RangeIndexReader.java

```diff
@@ -26,6 +26,51 @@
  * @param <T>
  */
 public interface RangeIndexReader<T> extends Closeable {
+
+  /**
+   * @return true if the results are exact and don't need refinement by scanning.
+   * This means {@see getPartiallyMatchingDocIds} will return null.
+   */
+  default boolean isExact() {
+    return true;
```

Comment:
I would not make this a default. shouldn't the implementer decide?

Reply:
There are two implementations, the first one was inexact and could be up to 10x slower than the one which is exact. If we implement another range index, we will want it to be exact.

Reply:
oh yes. what I meant is we should not implement a default --> whether the implementation is exact or not cannot be determined in the interface level - e.g. if I accidentally deleted the override in `RangeIndexReaderImpl` i will get no compile time error until i run into test failures. 

is the intent here to not allow any non-exact impl in the future?

Reply:
> is the intent here to not allow any non-exact impl in the future?

exactly, you need to opt out of being exact and have a good reason